clippy_lints/src/types.rs

```diff
@@ -25,7 +25,7 @@ use rustc::hir::intravisit::{walk_body, walk_expr, walk_ty, FnKind, NestedVisito
 use rustc::hir::*;
 use rustc::lint::{in_external_macro, LateContext, LateLintPass, LintArray, LintContext, LintPass};
 use rustc::ty::layout::LayoutOf;
-use rustc::ty::{self, Ty, TyCtxt, TypeckTables};
+use rustc::ty::{self, Ref, Ty, TyCtxt, TypeckTables};
```

Comment:
Ah, these are normally not imported like this, and accessed with `ty::Ref` instead (using the `ty::self` import).